sherpa/astro/io/pyfits_backend.py

```diff
@@ -65,6 +90,23 @@
     _new_header = fits.CardList
 
 
+# As fits.new_table is deprecated, use the from_columns method
+# if available. Note that this should only be done for binary
+# tables, as fits.TableHDU.from_columns should be used for
+# ASCII tables.
+#
+if hasattr(fits.BinTableHDU, "from_columns"):
+    _new_table = fits.BinTableHDU.from_columns
+else:
+    _new_table = fits.new_table
+
+# fits.CardList is deprecated
```

Comment:
This (the definition of `_new_header`) isn't needed as it is already at line 86.